src/Core/View.js

```diff
@@ -629,4 +632,96 @@ View.prototype.pickObjectsAt = function pickObjectsAt(mouseOrEvt, ...where) {
     return results;
 };
 
+function viewEvent(type, event, view) {
+    this.type = type;
+    this.viewCoords = view.eventToViewCoords(event).clone();
+    let normalizedCoords;
+    let pickeds;
+    let coordinate;
+    Object.defineProperty(
+        this,
+        'normalizedCoords',
+        {
+            get: () => {
+                normalizedCoords = normalizedCoords || view.viewToNormalizedCoords(this.viewCoords);
+                return normalizedCoords;
+            },
+        });
+    Object.defineProperty(
+    this,
+    'pickeds',
+        {
+            get: () => {
+                pickeds = pickeds || view.pickObjectsAt(event, ...this.where);
+                return pickeds;
+            },
+        });
+    Object.defineProperty(
+    this,
+    'coordinate',
+        {
+            get: () => {
+                const tileLayer = view.tileLayer || view.wgs84TileLayer;
+                const pickedPosition = view.getPickingPositionFromDepth(this.viewCoords);
+                if (pickedPosition) {
+                    coordinate = coordinate || new Coordinates(view.referenceCrs, pickedPosition).as(tileLayer.extent.crs());
+                }
+                return coordinate;
+            },
+        });
+}
+
+/*
+// Another viewEvent implementation
+const _coordinate = new WeakMap();
+class viewEventB {
+    constructor(type, event, view) {
+        this.type = type;
+        this.viewCoords = view.eventToViewCoords(event).clone();
+        this.view = view;
+        _coordinate.set(this, null);
+    }
+    get coordinate() {
+        _coordinate.set(this);
+        const tileLayer = this.view.tileLayer || this.view.wgs84TileLayer;
+        const pickedPosition = this.view.getPickingPositionFromDepth(this.viewCoords);
+        if (pickedPosition) {
+            _coordinate.set(this, _coordinate.get(this) || new Coordinates(this.view.referenceCrs, pickedPosition).as(tileLayer.extent.crs()));
+        }
+        return _coordinate.get(this);
+    }
+}
+*/
+
+const domListeners = {};
+const viewListeners = {};
+View.prototype.on = function _on(type, listener, ...where) {
+    if (!domListeners[type]) {
+        const domElement = this.mainLoop.gfxEngine.renderer.domElement;
+        domListeners[type] = (event) => {
+            this.dispatchEvent(new viewEvent(type, event, this));
+        };
+        domElement.addEventListener(type, domListeners[type], false);
+    }
+
+    const viewerListener = (event) => {
+        event.where = where;
+        listener(event);
+    };
+
+    viewListeners[listener] = viewerListener;
+    this.addEventListener(type, viewerListener);
+};
```

Comment:
This is not the correct approach. Imho the interesting part of this PR is the autocalculation of `view.eventToViewCoords`, but in this case, you don't need a new object and this complicated double-event listeners. Just proxy and call user callback with your augmented event (you can add `viewX` and `viewY` properties to the MouseEvent for examples).